cpumanager e2e test fails on multi-node NUMA system

**Which jobs are failing**:
N/A - AFAICT CI jobs works fine

**Which test(s) are failing**:
```bash
make test-e2e-node PARALLELISM=1 FOCUS="Feature:CPUManager" SKIP=""
...

STEP: running a Gu pod requesting multiple CPUs
STEP: checking if the expected cpuset was assigned
[AfterEach] [sig-node] CPU Manager [Serial] [Feature:CPUManager][NodeAlphaFeature:CPUManager]
  /home/fromani/proj/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:178
STEP: Collecting events from namespace "cpu-manager-test-3145".
STEP: Found 19 events.
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:09:22 -0500 EST - event for non-gu-pod: {kubelet multinuma.example.host} Pulled: Container image "docker.io/library/busybox:1.29" already present on machine
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:09:23 -0500 EST - event for non-gu-pod: {kubelet multinuma.example.host} Started: Started container non-gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:09:23 -0500 EST - event for non-gu-pod: {kubelet multinuma.example.host} Created: Created container non-gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:09:24 -0500 EST - event for non-gu-pod: {kubelet multinuma.example.host} Killing: Stopping container non-gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:10:21 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Pulled: Container image "docker.io/library/busybox:1.29" already present on machine
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:10:22 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Created: Created container gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:10:22 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Started: Started container gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:10:23 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Killing: Stopping container gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:11:22 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Created: Created container gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:11:22 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Started: Started container gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:11:22 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Pulled: Container image "docker.io/library/busybox:1.29" already present on machine
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:11:24 -0500 EST - event for non-gu-pod: {kubelet multinuma.example.host} Pulled: Container image "docker.io/library/busybox:1.29" already present on machine
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:11:24 -0500 EST - event for non-gu-pod: {kubelet multinuma.example.host} Created: Created container non-gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:11:24 -0500 EST - event for non-gu-pod: {kubelet multinuma.example.host} Started: Started container non-gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:11:25 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Killing: Stopping container gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:11:26 -0500 EST - event for non-gu-pod: {kubelet multinuma.example.host} Killing: Stopping container non-gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:12:22 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Started: Started container gu-container
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:12:22 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Pulled: Container image "docker.io/library/busybox:1.29" already present on machine
Jan 31 08:12:23.892: INFO: At 2020-01-31 08:12:22 -0500 EST - event for gu-pod: {kubelet multinuma.example.host} Created: Created container gu-container
Jan 31 08:12:23.893: INFO: POD     NODE                                   PHASE    GRACE  CONDITIONS
Jan 31 08:12:23.893: INFO: gu-pod  multinuma.example.host  Running         [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2020-01-31 08:12:21 -0500 EST  } {Ready True 0001-01-01 00:00:00 +0000 UTC 2020-01-31 08:12:23 -0500 EST  } {ContainersReady True 0001-01-01 00:00:00 +0000 UTC 2020-01-31 08:12:23 -0500 EST  } {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2020-01-31 08:12:21 -0500 EST  }]
Jan 31 08:12:23.893: INFO: 
Jan 31 08:12:23.895: INFO: 
Logging node info for node multinuma.example.host
Jan 31 08:12:23.897: INFO: Node Info: &Node{ObjectMeta:{multinuma.example.host   /api/v1/nodes/multinuma.example.host dd2f2200-f481-4336-b9d0-2cc94d84fe6a 94 0 2020-01-31 08:07:56 -0500 EST <nil> <nil> map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:multinuma.example.host kubernetes.io/os:linux] map[volumes.kubernetes.io/controller-managed-attach-detach:true] [] []  [{e2e_node.test Update v1 2020-01-31 08:08:07 -0500 EST FieldsV1 FieldsV1{Raw:*[123 34 102 58 115 112 101 99 34 58 123 34 102 58 99 111 110 102 105 103 83 111 117 114 99 101 34 58 123 34 46 34 58 123 125 44 34 102 58 99 111 110 102 105 103 77 97 112 34 58 123 34 46 34 58 123 125 44 34 102 58 107 117 98 101 108 101 116 67 111 110 102 105 103 75 101 121 34 58 123 125 44 34 102 58 110 97 109 101 34 58 123 125 44 34 102 58 110 97 109 101 115 112 97 99 101 34 58 123 125 125 125 125 125],}} {kubelet Update v1 2020-01-31 08:09:25 -0500 EST FieldsV1 &FieldsV1{Raw:*[123 34 102 58 109 101 116 97 100 97 116 97 34 58 123 34 102 58 97 110 110 111 116 97 116 105 111 110 115 34 58 123 34 46 34 58 123 125 44 34 102 58 118 111 108 117 109 101 115 46 107 117 98 101 114 110 101 116 101 115 46 105 111 47 99 111 110 116 114 111 108 108 101 114 45 109 97 110 97 103 101 100 45 97 116 116 97 99 104 45 100 101 116 97 99 104 34 58 123 125 125 44 34 102 58 108 97 98 101 108 115 34 58 123 34 46 34 58 123 125 44 34 102 58 98 101 116 97 46 107 117 98 101 114 110 101 116 101 115 46 105 111 47 97 114 99 104 34 58 123 125 44 34 102 58 98 101 116 97 46 107 117 98 101 114 110 101 116 101 115 46 105 111 47 111 115 34 58 123 125 44 34 102 58 107 117 98 101 114 110 101 116 101 115 46 105 111 47 97 114 99 104 34 58 123 125 44 34 102 58 107 117 98 101 114 110 101 116 101 115 46 105 111 47 104 111 115 116 110 97 109 101 34 58 123 125 44 34 102 58 107 117 98 101 114 110 101 116 101 115 46 105 111 47 111 115 34 58 123 125 125 125 44 34 102 58 115 116 97 116 117 115 34 58 123 34 102 58 97 100 100 114 101 115 115 101 115 34 58 123 34 46 34 58 123 125 44 34 107 58 123 92 34 116 121 112 101 92 34 58 92 34 72 111 115 116 110 97 109 101 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 97 100 100 114 101 115 115 34 58 123 125 44 34 102 58 116 121 112 101 34 58 123 125 125 44 34 107 58 123 92 34 116 121 112 101 92 34 58 92 34 73 110 116 101 114 110 97 108 73 80 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 97 100 100 114 101 115 115 34 58 123 125 44 34 102 58 116 121 112 101 34 58 123 125 125 125 44 34 102 58 97 108 108 111 99 97 116 97 98 108 101 34 58 123 34 46 34 58 123 125 44 34 102 58 99 112 117 34 58 123 125 44 34 102 58 101 112 104 101 109 101 114 97 108 45 115 116 111 114 97 103 101 34 58 123 125 44 34 102 58 104 117 103 101 112 97 103 101 115 45 49 71 105 34 58 123 125 44 34 102 58 104 117 103 101 112 97 103 101 115 45 50 77 105 34 58 123 125 44 34 102 58 105 110 116 101 108 46 99 111 109 47 105 110 116 101 108 95 115 114 105 111 118 95 100 112 100 107 34 58 123 125 44 34 102 58 105 110 116 101 108 46 99 111 109 47 105 110 116 101 108 95 115 114 105 111 118 95 110 101 116 100 101 118 105 99 101 34 58 123 125 44 34 102 58 105 110 116 101 108 46 99 111 109 47 109 108 110 120 95 115 114 105 111 118 95 114 100 109 97 34 58 123 125 44 34 102 58 109 101 109 111 114 121 34 58 123 125 44 34 102 58 112 111 100 115 34 58 123 125 125 44 34 102 58 99 97 112 97 99 105 116 121 34 58 123 34 46 34 58 123 125 44 34 102 58 99 112 117 34 58 123 125 44 34 102 58 101 112 104 101 109 101 114 97 108 45 115 116 111 114 97 103 101 34 58 123 125 44 34 102 58 104 117 103 101 112 97 103 101 115 45 49 71 105 34 58 123 125 44 34 102 58 104 117 103 101 112 97 103 101 115 45 50 77 105 34 58 123 125 44 34 102 58 105 110 116 101 108 46 99 111 109 47 105 110 116 101 108 95 115 114 105 111 118 95 100 112 100 107 34 58 123 125 44 34 102 58 105 110 116 101 108 46 99 111 109 47 105 110 116 101 108 95 115 114 105 111 118 95 110 101 116 100 101 118 105 99 101 34 58 123 125 44 34 102 58 105 110 116 101 108 46 99 111 109 47 109 108 110 120 95 115 114 105 111 118 95 114 100 109 97 34 58 123 125 44 34 102 58 109 101 109 111 114 121 34 58 123 125 44 34 102 58 112 111 100 115 34 58 123 125 125 44 34 102 58 99 111 110 100 105 116 105 111 110 115 34 58 123 34 46 34 58 123 125 44 34 107 58 123 92 34 116 121 112 101 92 34 58 92 34 68 105 115 107 80 114 101 115 115 117 114 101 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 108 97 115 116 72 101 97 114 116 98 101 97 116 84 105 109 101 34 58 123 125 44 34 102 58 108 97 115 116 84 114 97 110 115 105 116 105 111 110 84 105 109 101 34 58 123 125 44 34 102 58 109 101 115 115 97 103 101 34 58 123 125 44 34 102 58 114 101 97 115 111 110 34 58 123 125 44 34 102 58 115 116 97 116 117 115 34 58 123 125 44 34 102 58 116 121 112 101 34 58 123 125 125 44 34 107 58 123 92 34 116 121 112 101 92 34 58 92 34 77 101 109 111 114 121 80 114 101 115 115 117 114 101 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 108 97 115 116 72 101 97 114 116 98 101 97 116 84 105 109 101 34 58 123 125 44 34 102 58 108 97 115 116 84 114 97 110 115 105 116 105 111 110 84 105 109 101 34 58 123 125 44 34 102 58 109 101 115 115 97 103 101 34 58 123 125 44 34 102 58 114 101 97 115 111 110 34 58 123 125 44 34 102 58 115 116 97 116 117 115 34 58 123 125 44 34 102 58 116 121 112 101 34 58 123 125 125 44 34 107 58 123 92 34 116 121 112 101 92 34 58 92 34 80 73 68 80 114 101 115 115 117 114 101 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 108 97 115 116 72 101 97 114 116 98 101 97 116 84 105 109 101 34 58 123 125 44 34 102 58 108 97 115 116 84 114 97 110 115 105 116 105 111 110 84 105 109 101 34 58 123 125 44 34 102 58 109 101 115 115 97 103 101 34 58 123 125 44 34 102 58 114 101 97 115 111 110 34 58 123 125 44 34 102 58 115 116 97 116 117 115 34 58 123 125 44 34 102 58 116 121 112 101 34 58 123 125 125 44 34 107 58 123 92 34 116 121 112 101 92 34 58 92 34 82 101 97 100 121 92 34 125 34 58 123 34 46 34 58 123 125 44 34 102 58 108 97 115 116 72 101 97 114 116 98 101 97 116 84 105 109 101 34 58 123 125 44 34 102 58 108 97 115 116 84 114 97 110 115 105 116 105 111 110 84 105 109 101 34 58 123 125 44 34 102 58 109 101 115 115 97 103 101 34 58 123 125 44 34 102 58 114 101 97 115 111 110 34 58 123 125 44 34 102 58 115 116 97 116 117 115 34 58 123 125 44 34 102 58 116 121 112 101 34 58 123 125 125 125 44 34 102 58 99 111 110 102 105 103 34 58 123 34 46 34 58 123 125 44 34 102 58 97 99 116 105 118 101 34 58 123 34 46 34 58 123 125 44 34 102 58 99 111 110 102 105 103 77 97 112 34 58 123 34 46 34 58 123 125 44 34 102 58 107 117 98 101 108 101 116 67 111 110 102 105 103 75 101 121 34 58 123 125 44 34 102 58 110 97 109 101 34 58 123 125 44 34 102 58 110 97 109 101 115 112 97 99 101 34 58 123 125 44 34 102 58 114 101 115 111 117 114 99 101 86 101 114 115 105 111 110 34 58 123 125 44 34 102 58 117 105 100 34 58 123 125 125 125 44 34 102 58 97 115 115 105 103 110 101 100 34 58 123 34 46 34 58 123 125 44 34 102 58 99 111 110 102 105 103 77 97 112 34 58 123 34 46 34 58 123 125 44 34 102 58 107 117 98 101 108 101 116 67 111 110 102 105 103 75 101 121 34 58 123 125 44 34 102 58 110 97 109 101 34 58 123 125 44 34 102 58 110 97 109 101 115 112 97 99 101 34 58 123 125 44 34 102 58 114 101 115 111 117 114 99 101 86 101 114 115 105 111 110 34 58 123 125 44 34 102 58 117 105 100 34 58 123 125 125 125 125 44 34 102 58 100 97 101 109 111 110 69 110 100 112 111 105 110 116 115 34 58 123 34 102 58 107 117 98 101 108 101 116 69 110 100 112 111 105 110 116 34 58 123 34 102 58 80 111 114 116 34 58 123 125 125 125 44 34 102 58 105 109 97 103 101 115 34 58 123 125 44 34 102 58 110 111 100 101 73 110 102 111 34 58 123 34 102 58 97 114 99 104 105 116 101 99 116 117 114 101 34 58 123 125 44 34 102 58 98 111 111 116 73 68 34 58 123 125 44 34 102 58 99 111 110 116 97 105 110 101 114 82 117 110 116 105 109 101 86 101 114 115 105 111 110 34 58 123 125 44 34 102 58 107 101 114 110 101 108 86 101 114 115 105 111 110 34 58 123 125 44 34 102 58 107 117 98 101 80 114 111 120 121 86 101 114 115 105 111 110 34 58 123 125 44 34 102 58 107 117 98 101 108 101 116 86 101 114 115 105 111 110 34 58 123 125 44 34 102 58 109 97 99 104 105 110 101 73 68 34 58 123 125 44 34 102 58 111 112 101 114 97 116 105 110 103 83 121 115 116 101 109 34 58 123 125 44 34 102 58 111 115 73 109 97 103 101 34 58 123 125 44 34 102 58 115 121 115 116 101 109 85 85 73 68 34 58 123 125 125 125 125],}}]},Spec:NodeSpec{PodCIDR:,DoNotUseExternalID:,ProviderID:,Unschedulable:false,Taints:[]Taint{},ConfigSource:&NodeConfigSource{ConfigMap:&ConfigMapNodeConfigSource{Namespace:kube-system,Name:testcfg-mfrxv,UID:,ResourceVersion:,KubeletConfigKey:kubelet,},},PodCIDRs:[],},Status:NodeStatus{Capacity:ResourceList{cpu: {{24 0} {<nil>} 24 DecimalSI},ephemeral-storage: {{53660876800 0} {<nil>} 51175Mi BinarySI},hugepages-1Gi: {{0 0} {<nil>} 0 DecimalSI},hugepages-2Mi: {{0 0} {<nil>} 0 DecimalSI},intel.com/intel_sriov_dpdk: {{0 0} {<nil>} 0 DecimalSI},intel.com/intel_sriov_netdevice: {{0 0} {<nil>} 0 DecimalSI},intel.com/mlnx_sriov_rdma: {{0 0} {<nil>} 0 DecimalSI},memory: {{67333857280 0} {<nil>}  BinarySI},pods: {{110 0} {<nil>} 110 DecimalSI},},Allocatable:ResourceList{cpu: {{23800 -3} {<nil>} 23800m DecimalSI},ephemeral-storage: {{48294789041 0} {<nil>} 48294789041 DecimalSI},hugepages-1Gi: {{0 0} {<nil>} 0 DecimalSI},hugepages-2Mi: {{0 0} {<nil>} 0 DecimalSI},intel.com/intel_sriov_dpdk: {{0 0} {<nil>} 0 DecimalSI},intel.com/intel_sriov_netdevice: {{0 0} {<nil>} 0 DecimalSI},intel.com/mlnx_sriov_rdma: {{0 0} {<nil>} 0 DecimalSI},memory: {{67071713280 0} {<nil>}  BinarySI},pods: {{110 0} {<nil>} 110 DecimalSI},},Phase:,Conditions:[]NodeCondition{NodeCondition{Type:MemoryPressure,Status:False,LastHeartbeatTime:2020-01-31 08:09:25 -0500 EST,LastTransitionTime:2020-01-31 08:07:53 -0500 EST,Reason:KubeletHasSufficientMemory,Message:kubelet has sufficient memory available,},NodeCondition{Type:DiskPressure,Status:False,LastHeartbeatTime:2020-01-31 08:09:25 -0500 EST,LastTransitionTime:2020-01-31 08:07:53 -0500 EST,Reason:KubeletHasNoDiskPressure,Message:kubelet has no disk pressure,},NodeCondition{Type:PIDPressure,Status:False,LastHeartbeatTime:2020-01-31 08:09:25 -0500 EST,LastTransitionTime:2020-01-31 08:07:53 -0500 EST,Reason:KubeletHasSufficientPID,Message:kubelet has sufficient PID available,},NodeCondition{Type:Ready,Status:True,LastHeartbeatTime:2020-01-31 08:09:25 -0500 EST,LastTransitionTime:2020-01-31 08:08:06 -0500 EST,Reason:KubeletReady,Message:kubelet is posting ready status,},},Addresses:[]NodeAddress{NodeAddress{Type:InternalIP,Address:10.8.125.16,},NodeAddress{Type:Hostname,Address:multinuma.example.host,},},DaemonEndpoints:NodeDaemonEndpoints{KubeletEndpoint:DaemonEndpoint{Port:10250,},},NodeInfo:NodeSystemInfo{MachineID:9693047527f740bfa63f9332e62b48d8,SystemUUID:4C4C4544-0032-3710-8030-C4C04F4D3032,BootID:bf25519c-ea01-4e0b-b505-fa8896d6ff0f,KernelVersion:3.10.0-1062.7.1.el7.x86_64,OSImage:CentOS Linux 7 (Core),ContainerRuntimeVersion:docker://19.3.5,KubeletVersion:v1.18.0-alpha.2.305+1bb68a2cde08e3,KubeProxyVersion:v1.18.0-alpha.2.305+1bb68a2cde08e3,OperatingSystem:linux,Architecture:amd64,},Images:[]ContainerImage{ContainerImage{Names:[perl@sha256:b0286e5022897214b275e63c2169032de1e262579ead5b84fc5c801b3c71070b perl:5.26],SizeBytes:852899445,},ContainerImage{Names:[perl@sha256:e24087f67fed4193c711688df09cf5ff6752016c65ecbdd6717ad1bc0471f49c],SizeBytes:852897180,},ContainerImage{Names:[perl@sha256:6cef3addb526f33b0bb7f707721a95e4be2afa4b2731afc74c4a2a5dcb9cc2b3],SizeBytes:852880194,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/node-perf/tf-wide-deep-amd64@sha256:80d4564d5ab49ecfea3b20f75cc676d8dfd8b2aca364ed4c1a8a55fbcaaed7f6 gcr.io/kubernetes-e2e-test-images/node-perf/tf-wide-deep-amd64:1.0],SizeBytes:634132274,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/volume/gluster@sha256:e2d3308b2d27499d59f120ff46dfc6c4cb307a3f207f02894ecab902583761c9 gcr.io/kubernetes-e2e-test-images/volume/gluster:1.0],SizeBytes:331951020,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/volume/nfs@sha256:c2ad734346f608a5f7d69cfded93c4e8094069320657bd372d12ba21dea3ea71 gcr.io/kubernetes-e2e-test-images/volume/nfs:1.0],SizeBytes:225350895,},ContainerImage{Names:[httpd@sha256:eb8ccf084cf3e80eece1add239effefd171eb39adbc154d33c14260d905d4060 httpd:2.4.38-alpine],SizeBytes:123774598,},ContainerImage{Names:[k8s.gcr.io/node-problem-detector@sha256:6e9b4a4eaa47f120be61f60573a545844de63401661812e2cfb7ae81a28efd19 k8s.gcr.io/node-problem-detector:v0.6.2],SizeBytes:98706337,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/node-perf/npb-is@sha256:9d08dd99565b25af37c990cd4474a4284b27e7ceb3f98328bb481edefedf8aa5 gcr.io/kubernetes-e2e-test-images/node-perf/npb-is:1.0],SizeBytes:96270258,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/node-perf/npb-ep@sha256:564314549347619cfcdbe6c7d042a29e133a00e922b37682890fff17ac1a7804 gcr.io/kubernetes-e2e-test-images/node-perf/npb-ep:1.0],SizeBytes:96268458,},ContainerImage{Names:[google/cadvisor@sha256:815386ebbe9a3490f38785ab11bda34ec8dacf4634af77b8912832d4f85dca04 google/cadvisor:latest],SizeBytes:69559183,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/agnhost@sha256:daf5332100521b1256d0e3c56d697a238eaec3af48897ed9167cbadd426773b5 gcr.io/kubernetes-e2e-test-images/agnhost:2.8],SizeBytes:52778388,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/nonroot@sha256:4bd7ae247de5c988700233c5a4b55e804ffe90f8c66ae64853f1dae37b847213 gcr.io/kubernetes-e2e-test-images/nonroot:1.0],SizeBytes:42320060,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/nonroot@sha256:d4ede5c74517090b6686219059118ed178cf4620f5db8781b32f806bb1e7395b],SizeBytes:42320060,},ContainerImage{Names:[nfvpe/sriov-device-plugin@sha256:518499ed631ff84b43153b8f7624c1aaacb75a721038857509fe690abdf62ddb nfvpe/sriov-device-plugin:v3.1],SizeBytes:25314160,},ContainerImage{Names:[k8s.gcr.io/nvidia-gpu-device-plugin@sha256:4b036e8844920336fa48f36edeb7d4398f426d6a934ba022848deed2edbf09aa],SizeBytes:18977273,},ContainerImage{Names:[nginx@sha256:485b610fefec7ff6c463ced9623314a04ed67e3945b9c08d7e53a47f6d108dc7 nginx:1.14-alpine],SizeBytes:16028126,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/ipc-utils@sha256:bb127be3a1ecac0516f672a5e223d94fe6021021534ecb7a02a607a63154c3d8 gcr.io/kubernetes-e2e-test-images/ipc-utils:1.0],SizeBytes:10034171,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/nonewprivs@sha256:10066e9039219449fe3c81f38fe01928f87914150768ab81b62a468e51fa7411 gcr.io/kubernetes-e2e-test-images/nonewprivs:1.0],SizeBytes:6753187,},ContainerImage{Names:[k8s.gcr.io/stress@sha256:f00aa1ddc963a3164aef741aab0fc05074ea96de6cd7e0d10077cf98dd72d594 k8s.gcr.io/stress:v1],SizeBytes:5494760,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/test-webserver@sha256:7f93d6e32798ff28bc6289254d0c2867fe2c849c8e46edc50f8624734309812e gcr.io/kubernetes-e2e-test-images/test-webserver:1.0],SizeBytes:4732240,},ContainerImage{Names:[gcr.io/authenticated-image-pulling/alpine@sha256:7ff177862cb50c602bfe81f805969412e619c054a2bbead977d0c276988aa4a0 gcr.io/authenticated-image-pulling/alpine:3.7],SizeBytes:4202236,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/mounttest@sha256:c0bd6f0755f42af09a68c9a47fb993136588a76b3200ec305796b60d629d85d2 gcr.io/kubernetes-e2e-test-images/mounttest:1.0],SizeBytes:1563521,},ContainerImage{Names:[gcr.io/kubernetes-e2e-test-images/mounttest-user@sha256:17319ca525ee003681fccf7e8c6b1b910ff4f49b653d939ac7f9b6e7c463933d gcr.io/kubernetes-e2e-test-images/mounttest-user:1.0],SizeBytes:1450451,},ContainerImage{Names:[busybox@sha256:8ccbac733d19c0dd4d70b4f0c1e12245b5fa3ad24758a11035ee505c629c0796 busybox:1.29],SizeBytes:1154361,},ContainerImage{Names:[k8s.gcr.io/busybox@sha256:4bdd623e848417d96127e16037743f0cd8b528c026e9175e22a84f639eca58ff],SizeBytes:1113554,},ContainerImage{Names:[k8s.gcr.io/pause@sha256:f78411e19d84a252e53bff71a4407a5686c46983a2c2eeed83929b888179acea k8s.gcr.io/pause:3.1],SizeBytes:742472,},},VolumesInUse:[],VolumesAttached:[]AttachedVolume{},Config:&NodeConfigStatus{Assigned:&NodeConfigSource{ConfigMap:&ConfigMapNodeConfigSource{Namespace:kube-system,Name:testcfg-mfrxv,UID:3074fe07-a993-4164-8629-46ddc05c2396,ResourceVersion:61,KubeletConfigKey:kubelet,},},Active:&NodeConfigSource{ConfigMap:&ConfigMapNodeConfigSource{Namespace:kube-system,Name:testcfg-mfrxv,UID:3074fe07-a993-4164-8629-46ddc05c2396,ResourceVersion:61,KubeletConfigKey:kubelet,},},LastKnownGood:nil,Error:,},},}
Jan 31 08:12:23.898: INFO: 
Logging kubelet events for node multinuma.example.host
Jan 31 08:12:23.899: INFO: 
Logging pods the kubelet thinks is on node multinuma.example.host
Jan 31 08:12:23.901: INFO: gu-pod started at 2020-01-31 08:12:21 -0500 EST (0+1 container statuses recorded)
Jan 31 08:12:23.901: INFO: 	Container gu-container ready: true, restart count 0
W0131 08:12:23.902865    8561 metrics_grabber.go:80] Master node is not registered. Grabbing metrics from Scheduler, ControllerManager and ClusterAutoscaler is disabled.
Jan 31 08:12:23.915: INFO: 
Latency metrics for node multinuma.example.host
Jan 31 08:12:23.915: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
STEP: Destroying namespace "cpu-manager-test-3145" for this suite.

• Failure [256.700 seconds]
[sig-node] CPU Manager [Serial] [Feature:CPUManager][NodeAlphaFeature:CPUManager]
_output/local/go/src/k8s.io/kubernetes/test/e2e_node/framework.go:23
  With kubeconfig updated with static CPU Manager policy run the CPU Manager tests
  _output/local/go/src/k8s.io/kubernetes/test/e2e_node/cpu_manager_test.go:530
    should assign CPUs as expected based on the Pod spec [It]
    _output/local/go/src/k8s.io/kubernetes/test/e2e_node/cpu_manager_test.go:255

    expected log not found in container [gu-container] of pod [gu-pod]
    Unexpected error:
        <*errors.errorString | 0xc0007bfb40>: {
            s: "failed to match regexp \"^1,13\\n$\" in output \"2,14\\n\"",
        }
        failed to match regexp "^1,13\n$" in output "2,14\n"
    occurred

    _output/local/go/src/k8s.io/kubernetes/test/e2e_node/cpu_manager_test.go:388
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSI0131 08:12:23.920017    8561 e2e_node_suite_test.go:216] Stopping node services...
I0131 08:12:23.920030    8561 server.go:257] Kill server "services"
I0131 08:12:23.920047    8561 server.go:294] Killing process 9345 (services) with -TERM
E0131 08:12:23.939846    8561 services.go:88] Failed to stop services: error stopping "services": wait: no child processes
I0131 08:12:23.939880    8561 server.go:257] Kill server "kubelet"
I0131 08:12:23.948729    8561 services.go:147] Fetching log files...
I0131 08:12:23.949031    8561 services.go:156] Get log file "kern.log" with journalctl command [-k].
I0131 08:12:23.952583    8561 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I0131 08:12:23.952584    8561 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I0131 08:12:23.952600    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc000f242a0, {CONNECTING <nil>}
I0131 08:12:23.952625    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc0001ef640, {CONNECTING <nil>}
I0131 08:12:23.952633    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc0001ef840, {CONNECTING <nil>}
I0131 08:12:23.952642    8561 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I0131 08:12:23.952657    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc00093e830, {CONNECTING <nil>}
I0131 08:12:23.952673    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc000f24150, {CONNECTING <nil>}
I0131 08:12:23.952707    8561 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I0131 08:12:23.952708    8561 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
W0131 08:12:23.952855    8561 clientconn.go:1208] grpc: addrConn.createTransport failed to connect to {/var/run/dockershim.sock  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused". Reconnecting...
W0131 08:12:23.952861    8561 clientconn.go:1208] grpc: addrConn.createTransport failed to connect to {/var/run/dockershim.sock  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused". Reconnecting...
W0131 08:12:23.952862    8561 clientconn.go:1208] grpc: addrConn.createTransport failed to connect to {/var/run/dockershim.sock  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused". Reconnecting...
W0131 08:12:23.952855    8561 clientconn.go:1208] grpc: addrConn.createTransport failed to connect to {/var/run/dockershim.sock  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused". Reconnecting...
W0131 08:12:23.952869    8561 clientconn.go:1208] grpc: addrConn.createTransport failed to connect to {/var/run/dockershim.sock  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused". Reconnecting...
I0131 08:12:23.952950    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc000f242a0, {TRANSIENT_FAILURE connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused"}
I0131 08:12:23.952953    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc0001ef840, {TRANSIENT_FAILURE connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused"}
I0131 08:12:23.952947    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc00093e830, {TRANSIENT_FAILURE connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused"}
I0131 08:12:23.953015    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc000f24150, {TRANSIENT_FAILURE connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused"}
I0131 08:12:23.953024    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc0001ef640, {TRANSIENT_FAILURE connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused"}
I0131 08:12:23.954315    8561 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I0131 08:12:23.954345    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc000456e50, {CONNECTING <nil>}
I0131 08:12:23.954364    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc0004570d0, {CONNECTING <nil>}
I0131 08:12:23.954373    8561 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
W0131 08:12:23.954414    8561 clientconn.go:1208] grpc: addrConn.createTransport failed to connect to {/var/run/dockershim.sock  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused". Reconnecting...
I0131 08:12:23.954464    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc000456e50, {TRANSIENT_FAILURE connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused"}
W0131 08:12:23.954484    8561 clientconn.go:1208] grpc: addrConn.createTransport failed to connect to {/var/run/dockershim.sock  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused". Reconnecting...
I0131 08:12:23.954561    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc0004570d0, {TRANSIENT_FAILURE connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused"}
I0131 08:12:23.955287    8561 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I0131 08:12:23.955291    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc00093ea00, {CONNECTING <nil>}
W0131 08:12:23.955352    8561 clientconn.go:1208] grpc: addrConn.createTransport failed to connect to {/var/run/dockershim.sock  <nil> 0 <nil>}. Err :connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused". Reconnecting...
I0131 08:12:23.955426    8561 balancer_conn_wrappers.go:78] pickfirstBalancer: HandleSubConnStateChange: 0xc00093ea00, {TRANSIENT_FAILURE connection error: desc = "transport: Error while dialing dial unix /var/run/dockershim.sock: connect: connection refused"}
I0131 08:12:24.006330    8561 services.go:156] Get log file "cloud-init.log" with journalctl command [-u cloud*].
E0131 08:12:24.013756    8561 services.go:159] failed to get "cloud-init.log" from journald: Failed to add filter for units: No data available
, exit status 1
I0131 08:12:24.013788    8561 services.go:156] Get log file "docker.log" with journalctl command [-u docker].
I0131 08:12:24.025386    8561 services.go:156] Get log file "kubelet.log" with journalctl command [-u kubelet-20200131T080745.service].
I0131 08:12:24.463267    8561 e2e_node_suite_test.go:221] Tests Finished

JUnit report was created: /tmp/_artifacts/200131T080647/junit__01.xml


Summarizing 1 Failure:

[Fail] [sig-node] CPU Manager [Serial] [Feature:CPUManager][NodeAlphaFeature:CPUManager] With kubeconfig updated with static CPU Manager policy run the CPU Manager tests [It] should assign CPUs as expected based on the Pod spec 
_output/local/go/src/k8s.io/kubernetes/test/e2e_node/cpu_manager_test.go:388

Ran 1 of 316 Specs in 294.697 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 315 Skipped
--- FAIL: TestE2eNode (294.70s)
FAIL

Ginkgo ran 1 suite in 4m55.535252306s
Test Suite Failed
F0131 08:12:24.480907    6796 run_local.go:70] Test failed: exit status 1
exit status 1
make: *** [test-e2e-node] Error 1
```

**Since when has it been failing**:
no historical record on CI

**Testgrid link**:
N/A

**Reason for failure**:
I believe the tests make too many assumptions around the hardware, and the code is actually OK. to be confirmed.

**Anything else we need to know**:
```bash
$ lscpu
Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                24
On-line CPU(s) list:   0-23
Thread(s) per core:    2
Core(s) per socket:    6
Socket(s):             2
NUMA node(s):          2
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 62
Model name:            Intel(R) Xeon(R) CPU E5-2620 v2 @ 2.10GHz
Stepping:              4
CPU MHz:               1215.344
CPU max MHz:           2600.0000
CPU min MHz:           1200.0000
BogoMIPS:              4200.02
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              15360K
NUMA node0 CPU(s):     0,2,4,6,8,10,12,14,16,18,20,22
NUMA node1 CPU(s):     1,3,5,7,9,11,13,15,17,19,21,23
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms xsaveopt dtherm ida arat pln pts md_clear spec_ctrl intel_stibp flush_l1d
```

--- a/test/e2e_node/cpu_manager_test.go
+++ b/test/e2e_node/cpu_manager_test.go
@@ -145,8 +145,24 @@ func isHTEnabled() bool {
 	return threadsPerCore > 1
 }
 
+func isMultiNUMA() bool {
+	outData, err := exec.Command("/bin/sh", "-c", "lscpu | grep \"NUMA node(s):\" | cut -d \":\" -f 2").Output()
+	framework.ExpectNoError(err)
+
+	numaNodes, err := strconv.Atoi(strings.TrimSpace(string(outData)))
+	framework.ExpectNoError(err)
+
+	return numaNodes > 1
+}
+
 func getCPUSiblingList(cpuRes int64) string {
 	out, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("cat /sys/devices/system/cpu/cpu%d/topology/thread_siblings_list | tr -d \"\n\r\"", cpuRes)).Output()
+	framework.ExpectNoError(err)
+	return string(out)
+}
+
+func getCoreSiblingList(cpuRes int64) string {
+	out, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("cat /sys/devices/system/cpu/cpu%d/topology/core_siblings_list | tr -d \"\n\r\"", cpuRes)).Output()
 	framework.ExpectNoError(err)
 	return string(out)
 }
@@ -301,6 +317,9 @@ func runCPUManagerTests(f *framework.Framework) {
 		if isHTEnabled() {
 			cpuList = cpuset.MustParse(getCPUSiblingList(0)).ToSlice()
 			cpu1 = cpuList[1]
+		} else if isMultiNUMA() {
+			cpuList = cpuset.MustParse(getCoreSiblingList(0)).ToSlice()
+			cpu1 = cpuList[1]
 		}
 		expAllowedCPUsListRegex = fmt.Sprintf("^%d\n$", cpu1)
 		err = f.PodClient().MatchContainerOutput(pod.Name, pod.Spec.Containers[0].Name, expAllowedCPUsListRegex)
@@ -336,6 +355,9 @@ func runCPUManagerTests(f *framework.Framework) {
 		cpu1 = 1
 		if isHTEnabled() {
 			cpuList = cpuset.MustParse(getCPUSiblingList(0)).ToSlice()
+			cpu1 = cpuList[1]
+		} else if isMultiNUMA() {
+			cpuList = cpuset.MustParse(getCoreSiblingList(0)).ToSlice()
 			cpu1 = cpuList[1]
 		}
 		expAllowedCPUsListRegex = fmt.Sprintf("^%d\n$", cpu1)
@@ -376,7 +398,15 @@ func runCPUManagerTests(f *framework.Framework) {
 
 		ginkgo.By("checking if the expected cpuset was assigned")
 		cpuListString = "1-2"
-		if isHTEnabled() {
+		if isMultiNUMA() {
+			cpuList = cpuset.MustParse(getCoreSiblingList(0)).ToSlice()
+			if !isHTEnabled() {
+				cset = cpuset.MustParse(fmt.Sprintf("%d,%d", cpuList[1], cpuList[2]))
+			} else {
+				cset = cpuset.MustParse(getCPUSiblingList(int64(cpuList[1])))
+			}
+			cpuListString = fmt.Sprintf("%s", cset)
+		} else if isHTEnabled() {
 			cpuListString = "2-3"
 			cpuList = cpuset.MustParse(getCPUSiblingList(0)).ToSlice()
 			if cpuList[1] != 1 {
@@ -416,8 +446,14 @@ func runCPUManagerTests(f *framework.Framework) {
 			if cpuList[1] != 1 {
 				cpu1, cpu2 = cpuList[1], 1
 			}
+			if isMultiNUMA() {
+				cpuList = cpuset.MustParse(getCoreSiblingList(0)).ToSlice()
+				cpu2 = cpuList[1]
+			}
+		} else if isMultiNUMA() {
+			cpuList = cpuset.MustParse(getCoreSiblingList(0)).ToSlice()
+			cpu1, cpu2 = cpuList[1], cpuList[2]
 		}
-
 		expAllowedCPUsListRegex = fmt.Sprintf("^%d|%d\n$", cpu1, cpu2)
 		err = f.PodClient().MatchContainerOutput(pod.Name, pod.Spec.Containers[0].Name, expAllowedCPUsListRegex)
 		framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
@@ -460,8 +496,14 @@ func runCPUManagerTests(f *framework.Framework) {
 			if cpuList[1] != 1 {
 				cpu1, cpu2 = cpuList[1], 1
 			}
+			if isMultiNUMA() {
+				cpuList = cpuset.MustParse(getCoreSiblingList(0)).ToSlice()
+				cpu2 = cpuList[1]
+			}
+		} else if isMultiNUMA() {
+			cpuList = cpuset.MustParse(getCoreSiblingList(0)).ToSlice()
+			cpu1, cpu2 = cpuList[1], cpuList[2]
 		}
-
 		expAllowedCPUsListRegex = fmt.Sprintf("^%d\n$", cpu1)
 		err = f.PodClient().MatchContainerOutput(pod1.Name, pod1.Spec.Containers[0].Name, expAllowedCPUsListRegex)
 		framework.ExpectNoError(err, "expected log not found in container [%s] of pod [%s]",
@@ -499,6 +541,9 @@ func runCPUManagerTests(f *framework.Framework) {
 		cpu1 = 1
 		if isHTEnabled() {
 			cpuList = cpuset.MustParse(getCPUSiblingList(0)).ToSlice()
+			cpu1 = cpuList[1]
+		} else if isMultiNUMA() {
+			cpuList = cpuset.MustParse(getCoreSiblingList(0)).ToSlice()
 			cpu1 = cpuList[1]
 		}
 		expAllowedCPUsListRegex = fmt.Sprintf("^%d\n$", cpu1)

